libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -415,4 +449,178 @@ void AP_Camera_Backend::log_picture()
+    video_stream_info.msg.count = 1;
+    video_stream_info.msg.flags = VIDEO_STREAM_STATUS_FLAGS_RUNNING;
+
+    // If we got this far, then we've got all the required fields.
+    video_stream_info.is_valid = true;
+
+err:
+    delete obj;
+
+    if (video_stream_info.is_valid) {
+        debug("Loaded video_stream_info from '%s'", filename);
+        debug("    video_stream_info.msg.stream_id=%u", video_stream_info.msg.stream_id);
+        debug("    video_stream_info.msg.count=%u", video_stream_info.msg.count);
+        debug("    video_stream_info.msg.type=%u", video_stream_info.msg.type);
+        debug("    video_stream_info.msg.flags=%u", video_stream_info.msg.flags);
+        debug("    video_stream_info.msg.framerate=%f", video_stream_info.msg.framerate);
+        debug("    video_stream_info.msg.resolution_h=%u", video_stream_info.msg.resolution_h);
+        debug("    video_stream_info.msg.resolution_v=%u", video_stream_info.msg.resolution_v);
+        debug("    video_stream_info.msg.bitrate=%u", video_stream_info.msg.bitrate);
+        debug("    video_stream_info.msg.rotation=%u", video_stream_info.msg.rotation);
+        debug("    video_stream_info.msg.hfov=%u", video_stream_info.msg.hfov);
+
+        debug("    video_stream_info.msg.name='%s'", video_stream_info.msg.name);
+        debug("    video_stream_info.msg.uri='%s'", video_stream_info.msg.uri);
+    }
+}
+
+void AP_Camera_Backend::init_camera_information_from_json()
+{
+    video_stream_info.is_valid = false;
+
+    const auto filename = "camera_information.json";
+    auto *obj = _load_mount_msg_json(filename);
+    if (obj == nullptr) {
+        return;
+    }
+
+    if (!_copy_json_field_double(obj, "firmware_version", camera_info.msg.firmware_version)) goto err;
+    if (!_copy_json_field_double(obj, "focal_length", camera_info.msg.focal_length)) goto err;
+    if (!_copy_json_field_double(obj, "sensor_size_h", camera_info.msg.sensor_size_h)) goto err;
+    if (!_copy_json_field_double(obj, "sensor_size_v", camera_info.msg.sensor_size_v)) goto err;
+    if (!_copy_json_field_double(obj, "resolution_h", camera_info.msg.resolution_h)) goto err;
+    if (!_copy_json_field_double(obj, "resolution_v", camera_info.msg.resolution_v)) goto err;
+    if (!_copy_json_field_double(obj, "flags", camera_info.msg.flags)) goto err;
+    if (!_copy_json_field_double(obj, "cam_definition_version", camera_info.msg.cam_definition_version)) goto err;
+
+    if (!_copy_json_field_string(obj, "vendor_name", (char*)camera_info.msg.vendor_name, sizeof(camera_info.msg.vendor_name))) goto err;
+    if (!_copy_json_field_string(obj, "model_name", (char*)camera_info.msg.model_name, sizeof(camera_info.msg.model_name))) goto err;
+    if (!_copy_json_field_string(obj, "cam_definition_uri", camera_info.msg.cam_definition_uri, sizeof(camera_info.msg.cam_definition_uri))) goto err;
+
+    // Populate the fields that shouldn't change.
+    camera_info.msg.time_boot_ms = AP_HAL::millis();
```

Comment:
Errr.. this one should definitely change

Reply:
Fixed.